llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -300,26 +293,35 @@ Constant *getConstantElement(Constant *C, unsigned i, Type *elementType) {
 void collectCompositeElementsDefaultValuesRecursive(
     const Module &M, Constant *C, unsigned &Offset,
     std::vector<char> &DefaultValues) {
-  Type *Ty = C->getType();
-  if (auto *DataSeqC = dyn_cast<ConstantDataSequential>(C)) {
+  if (isa<ConstantAggregateZero>(C)) {
+    // This code is generic for zeroinitializer for both arrays and structs
+    size_t NumBytes = M.getDataLayout().getTypeStoreSize(C->getType());
+    std::fill_n(std::back_inserter(DefaultValues), NumBytes, 0);
+    Offset += NumBytes;
+  } else if (auto *DataSeqC = dyn_cast<ConstantDataSequential>(C)) {
```

Comment:
```suggestion
    Offset += NumBytes;
    return;
  }
  
  if (auto *DataSeqC = dyn_cast<ConstantDataSequential>(C)) {
```
I suggest using early exit to make code more readable.

Reply:
Thank you for the suggestion, I've changes the code to use early exit after each case.